airflow/providers/amazon/aws/hooks/elasticache.py

```diff
@@ -0,0 +1,238 @@
+        self.initial_poke_interval = initial_poke_interval
+
+        super().__init__(client_type='elasticache', *args, **kwargs)
+
+    def __create_replication_group(self, config):
+        """
+        Call ElastiCache API for creating a replication group
+        :param config: Python dictionary to use as config for creating replication group
+        :return: Response from ElastiCache create replication group API
+        """
+        return self.get_conn().create_replication_group(**config)
+
+    def __delete_replication_group(self, replication_group_id):
+        """
+        Call ElastiCache API for deleting a replication group
+        :param replication_group_id: ID of replication group to delete
+        :return: Response from ElastiCache delete replication group API
+        """
+        return self.get_conn().delete_replication_group(ReplicationGroupId=replication_group_id)
+
+    def describe_replication_group(self, replication_group_id):
+        """
+        Call ElastiCache API for describing a replication group
+        :param replication_group_id: ID of replication group to describe
+        :return: Response from ElastiCache describe replication group API
+        """
+        return self.get_conn().describe_replication_groups(ReplicationGroupId=replication_group_id)
+
+    def create_replication_group(self, config):
+        """
+        Create replication group with given config and check if it being created or not
+        :param config: Python dictionary to use as config for creating replication group
+        :return: Response from ElastiCache create replication group API
+        """
+        self.log.info('Creating replication group with config : %s', config)
+
+        response = self.__create_replication_group(config=config)
+
+        status = response[ElastiCacheDescribeResponseKeys.REPLICATION_GROUP][
+            ElastiCacheDescribeResponseKeys.STATUS]
+
+        if status != ElastiCacheStatus.STATUS_CREATING:
+            raise AirflowException('Replication group could not be created. Status: "{0}"'.format(status))
+
+        return response
+
+    def _is_replication_group_available(self, replication_group_id):
+        """
+        Helper for checking if replication is available or not
+        :param replication_group_id: ID of replication group to check for availability
+        :return: Flag to check if availability check should be stopped or not and current status of
+        replication group
+        """
+        desc = self.describe_replication_group(replication_group_id)
+
+        status = desc[ElastiCacheDescribeResponseKeys.REPLICATION_GROUPS][0][
+            ElastiCacheDescribeResponseKeys.STATUS]
+
+        return status in (
+            ElastiCacheStatus.STATUS_AVAILABLE,
+            ElastiCacheStatus.STATUS_CREATE_FAILED,
+            ElastiCacheStatus.STATUS_DELETING
+        ), status
+
+    def is_replication_group_available(self, replication_group_id):
```

Comment:
should be called `wait_somerthing` since it polls/loops.

max time should be an argument to this fn too.